Mathlib/RingTheory/TensorProduct.lean

```diff
@@ -436,28 +436,19 @@ instance : AddMonoidWithOne (A ⊗[R] B) :=
 
 instance : AddCommMonoid (A ⊗[R] B) := by infer_instance
 
-instance instSemiring : Semiring (A ⊗[R] B) :=
-  { (by infer_instance : AddMonoidWithOne (A ⊗[R] B)),
-    (by infer_instance : AddCommMonoid (A ⊗[R] B)) with
-    zero := 0
-    add := (· + ·)
-    one := 1
-    mul := fun a b => mul a b
-    one_mul := one_mul
-    mul_one := mul_one
-    mul_assoc := mul_assoc
-    add_assoc := add_assoc
-    zero_add := zero_add
-    add_zero := add_zero
-    add_comm := add_comm
-    nsmul_succ := AddMonoid.nsmul_succ
-    natCast_succ := AddMonoidWithOne.natCast_succ
-    zero_mul := fun a => show mul 0 a = 0 by rw [map_zero, LinearMap.zero_apply]
-    mul_zero := fun a => show mul a 0 = 0 by rw [map_zero]
-    -- port note : `left_distrib` and `right_distrib` are proved by `simp` in mathlib3
-    left_distrib := fun a b c => show mul a (b + c) = mul a b + mul a c by rw [map_add]
-    right_distrib := fun a b c => show mul (a + b) c = mul a c + mul b c
-      by rw [map_add, LinearMap.add_apply] }
+-- note: we deliberately do not provide any fields that overlap with base class fields
+instance instSemiring : Semiring (A ⊗[R] B) where
+  -- porting note : `left_distrib` and `right_distrib` are proved by `simp` in mathlib3
+  -- See https://github.com/leanprover-community/mathlib4/issues/5026
+  -- Probably because `mul` is defined to be bi-R-linear and then coerced to function?
+  left_distrib a b c := show mul a (b + c) = mul a b + mul a c by simp
+  right_distrib a b c := show mul (a + b) c = mul a c + mul b c by simp
+  zero_mul a := show mul 0 a = 0 by simp
+  mul_zero a := show mul a 0 = 0 by simp
```

Comment:
Are we missing simp lemmas or something? Why might this have stopped working? Note that the type of `mul` is `LinearMap` rather than just a function. Note also that this was already an issue on master.

Reply:
This stopped working because these fields are about `HMul.hMul`, but the `mul` field we provide is `Mul.mul`. The goals can be solved with `simp [HMul.hMul]`. This probably comes up all over the place when building structures, and the answer is always going to be "you have to unfold `HFoo.foo` too".